### PR TITLE
Fix SonarQube build breaker test incorrectly failing CI build on success.

### PR DESCRIPTION
**Pull request by HSAR**

### CODE REVIEW

File: Tasks/Maven/CodeAnalysis/SonarQube/vsts-server-utils.ts [hunk 1]
```diff
@@ -103,6 +76,33 @@ export class VstsServerUtils {
     }
 
     /**
+     * If enabled, fails the build in response to a quality gate failure.
+     * @param sqRunSettings SonarQubeRunSettings object for the applicable run or null if this is a PR build and the object could not be created
+     * @param sqMetrics     SonarQube metrics for the applicable run
+     * @returns {Promise<void>} Promise resolved when action completes (NB: Setting build result to failed results in procees exit)
+     */
+    private static breakBuildIfQualityGateFails(sqRunSettings:SonarQubeRunSettings, sqMetrics:SonarQubeMetrics):Q.Promise<void> {
```
Comment (HSAR) [inline Tasks/Maven/CodeAnalysis/SonarQube/vsts-server-utils.ts]: A decision was made to move the test lower down. This has the effect of reducing code coverage, but this is considered acceptable because the loss in code coverage is limited to a single line _tl.setResult()_, for which test coverage offers minimal value anyway.

There is also a beneficial side effect of being able to return the _breakBuildIfQualityGateFails()_ to private, when it was previously exposed for testing.

File: Tasks/Maven/CodeAnalysis/SonarQube/vsts-server-utils.ts [hunk 2]
```diff
@@ -149,6 +149,11 @@ export class VstsServerUtils {
         return sqStagingDir;
     }
 
+    /**
+     * Utility method, returns true if the given string is undefined, null or has length 0
+     * @param str String to examine
+     * @returns {boolean}
+     */
```
Comment (HSAR) [inline Tasks/Maven/CodeAnalysis/SonarQube/vsts-server-utils.ts]: Unrelated change - added documentation.

File: Tests/L0/Maven/_suite.ts [hunk 3]
```diff
@@ -1791,15 +1776,9 @@ describe('Maven Suite', function () {
...
-            .then(() => {
-                // unhook captured functions since they are important system functions
-                capturedStream.unhook();
-                process.exit = capturedExit;
-
-                // Assert
-                assert(processExitInvoked == 0, `Expected process to have not exited. Actual: ${processExitInvoked}`);
-                return true;
+        return analysisMetrics.getTaskResultFromQualityGateStatus()
+            .then((taskResult) => {
```
Comment (HSAR) [inline Tests/L0/Maven/_suite.ts]: While it would be far better to use the actual TaskResult enum, referencing it triggers the lazy evaluation of further test code which fails unless the environment variable MOCK_RESPONSES is set (which both cannot and should not be done in this scenario).

File: Tasks/Gradle/CodeAnalysis/SonarQube/vsts-server-utils.ts [hunk 4]
```diff
@@ -53,20 +53,43 @@ export class VstsServerUtils {
     }
```
Comment (HSAR) [inline Tasks/Gradle/CodeAnalysis/SonarQube/vsts-server-utils.ts]: File is mirrored to match Maven/CodeAnalysis/SonarQube/vsts-server-utils.ts

**Comment by bgavrilMS** [comment]
LGTM
